deployment/k8s-config/scripts/harbor-backup.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+#https://github.com/restic/restic/releases/download/v0.12.1/restic_0.12.1_linux_amd64.bz2
+#https://restic.readthedocs.io/en/stable/index.html
+#https://github.com/stonezdj/harbor/blob/backup_restore/tools/harbor-backup.sh
+
+RESTIC_REPO='s3:repo-url'
+KEY_ID='<aws access key id>'
+ACCESS_KEY='<aws access key>'
+PASSWORD='<restic password>'
+
+POSTGRES_POD='harbor-postgresql-0'
+REDIS_POD='harbor-redis-master-0'
+REGISTRY_POD='harbor-registry'
+
+POSTGRES_DIR='/bitnami/postgresql'
+REDIS_DIR='/data'
+REGISTRY_DIR='/storage/docker/registry'
+
+PGPASSWORD='<postgresql password>'
+PGDUMP_DIR='/tmp/harbor'
+
+
+#***************************************************************
+#Backup Postgresql files
+#***************************************************************
+
+kubectl cp restic_0121 cadc-harbor/$POSTGRES_POD:/tmp/.
+kubectl exec -it --namespace=cadc-harbor $POSTGRES_POD -- bash -c "export AWS_ACCESS_KEY_ID=$KEY_ID ; export AWS_SECRET_ACCESS_KEY=$ACCESS_KEY ; export RESTIC_PASSWORD=$PASSWORD ; ./tmp/restic_0121 -r $RESTIC_REPO --verbose backup $POSTGRES_DIR"
+
+#***************************************************************
+#Backup Postgresql database dump (just to be extra safe)
+#***************************************************************
+
+kubectl exec -it --namespace=cadc-harbor $POSTGRES_POD -- bash -c "export PGPASSWORD=$PGPASSWORD ; mkdir -p $PGDUMP_DIR ; pg_dump -U postgres registry > $PGDUMP_DIR/registry.back ; pg_dump -U postgres postgres > $PGDUMP_DIR/postgres.back ; pg_dump -U postgres notarysigner > $PGDUMP_DIR/notarysigner.back ; pg_dump -U postgres notaryserver > $PGDUMP_DIR/notaryserver.back"
+kubectl exec -it --namespace=cadc-harbor $POSTGRES_POD -- bash -c "export AWS_ACCESS_KEY_ID=$KEY_ID ; export AWS_SECRET_ACCESS_KEY=$ACCESS_KEY ; export RESTIC_PASSWORD=$PASSWORD ; ./tmp/restic_0121 -r $RESTIC_REPO --verbose backup $PGDUMP_DIR"
+
+#***************************************************************
+#Backup Redis
+#***************************************************************
+
+kubectl cp restic_0121 cadc-harbor/$REDIS_POD:/tmp/.
```

Comment:
What and where is file `restic_0121`?

Reply:
I'll include the curl command to download the binary.

Reply:
I commited the restic amd64 linux binary since restic is released as a bzip2 binary and that isn't installed on the keel nodes by default.